File: AUGUST-2025/15-08-2025/p342.ex
defmodule Solution do
  @spec is_power_of_four(n :: integer) :: boolean
  def is_power_of_four(n) when n<=0, do: false
  def is_power_of_four(n) when n==1, do: true
  def is_power_of_four(n) do
    if rem(n,4)==0 do
        is_power_of_four(div(n,4))
    else 
        false
    end
  end
end

File: AUGUST-2025/15-08-2025/p342.swift
class Solution {
    func isPowerOfFour(_ n: Int) -> Bool {
        guard n > 0 else { return false }
        let power = logBase4(Double(n))
        return floor(power) == power ? true : false
    }

    func logBase4(_ value: Double) -> Double {
        return log(value) / log(4.0)
    }
}